llvm/include/llvm/CodeGen/MachineSSAUpdater.h

```diff
@@ -40,8 +40,8 @@ class MachineSSAUpdater {
   //typedef DenseMap<MachineBasicBlock*, Register> AvailableValsTy;
   void *AV = nullptr;
 
-  /// VRC - Register class of the current virtual register.
-  const TargetRegisterClass *VRC = nullptr;
+  /// RegAttrs - current virtual register, new registers copy its attributes.
+  Register RegAttrs;
```

Comment:
One downside is a TargetRegisterClass is truly supposed to be constant. The register's attributes could change in the background

Reply:
Do you have suggestion how to implement this part, I wanted to avoid virtual methods